Formula/func-e.rb

```diff
@@ -0,0 +1,35 @@
+class FuncE < Formula
+  desc "Easily run Envoy"
+  homepage "https://func-e.io"
+  url "https://github.com/tetratelabs/func-e/archive/v0.5.0.tar.gz"
+  sha256 "17fc6c5c0f7bea8ce59e0bfb315198cc9f9ecfda98d51909149d42ffd125d72f"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/tetratelabs/func-e/internal/version.funcE=#{version}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags.join(" "))
+  end
+
+  test do
+    func_e_home = testpath/".func-e"
+    ENV["FUNC_E_HOME"] = func_e_home
+
+    # While this says "--version", this is a legitimate test as the --version is interpreted by Envoy.
+    # Specifically, func-e downloads and installs Envoy. Finally, it runs `envoy --version`
```

Comment:
ps I could replicate what's done in envoy_rb also. mainly trying to keep the size down, and focus on responsibilities. As long as func-e can start envoy, we should expect envoy to be able to run config also. https://github.com/Homebrew/homebrew-core/blob/master/Formula/envoy.rb#L39-L53

If you want me to copy over some of this logic, I won't mind, but anyway that's the rationale!